### PR TITLE
Fix request context activation/deactivation in gRPC

### PR DESCRIPTION
**Pull request by Ladicek**

### CODE REVIEW

**Comment by Ladicek** [comment]
I'm not _totally_ sure if this is correct, but it _seems_ obvious. Would be happy to be proven wrong :-)

**Comment by Ladicek** [comment]
Interestingly, there's a couple of test failures :-) If noone beats me to it, I'll look on Monday.

**Comment by michalszynkiewicz** [comment]
I'll mark it ready for review to see the failures.

File: extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/context/GrpcRequestContextGrpcInterceptor.java [hunk 1]
```diff
@@ -74,6 +74,30 @@ public void onReady() {
...
+                            reqContext.deactivate();
+                        }
+                    }
+                }
+
+                @Override
+                public void onCancel() {
+                    boolean activated = activateContext();
+                    try {
+                        super.onHalfClose();
```
Comment (Ladicek) [inline extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/context/GrpcRequestContextGrpcInterceptor.java:93]:
```suggestion
                        super.onCancel();
```

:-)

Comment (michalszynkiewicz) [reply extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/context/GrpcRequestContextGrpcInterceptor.java:93]: unfortunately it doesn't work anyway, as I wrote on Zulip...

**Comment by Ladicek** [comment]
Closing, as @michalszynkiewicz is working on a much more complete fix.

**Pull request closed without merging**